test/manual/gfont.js

```diff
@@ -0,0 +1,182 @@
+      sandbox.verify();
+      sandbox.restore();
+    });
+
+    describe('fonts_load_custom_font_with_data', function() {
+      it('simply passes the data', function() {
+        expect(font).to.be.an('object');
+        expect(font).to.have.a.property('status').which.equals('loaded');
+        expect(font).to.have.a.property('data').which.equals('some data');
+        expect(font).to.have.a.property('captureCPointer').which.is.a('function');
+        expect(font).to.have.a.property('releaseCPointer').which.is.a('function');
+      });
+    });
+
+    describe('capture and release pointer', function() {
+      it('returns null if not loaded', function() {
+        font.status = 'loading';
+        sandbox.mock(rocky.module)
+          .expects('ccall').never();
+        var cPtr = font.captureCPointer();
+        expect(cPtr).to.equal(0);
+      });
+
+      it('captureCPointer', function() {
+        sandbox.mock(rocky.module.Runtime)
+          .expects('addFunction').twice()
+          .onCall(0).returns(123)
+          .onCall(1).returns(456);
+        var ccall = sandbox.mock(rocky.module)
+          .expects('ccall').twice();
+        ccall.onCall(0).returns(789)
+             .onCall(1).returns(321);
+
+        var cPtr = font.captureCPointer();
+        expect(cPtr).to.equal(321);
+        expect(font).to.have.a.property('read_cb').which.equals(123);
+        expect(font).to.have.a.property('get_size_cb').which.equals(456);
+        expect(font).to.have.a.property('resourceId').which.equals(789);
+
+        // sinon's mocks don't seem to work with multiple expectations on the same mock
+        // hence this check afterwards
+        expect(ccall.getCall(0).args).to.eql(
+          ['emx_resources_register_custom', 'number', ['number', 'number'], [123, 456]]);
+        expect(ccall.getCall(1).args).to.eql(
+          ['fonts_load_custom_font', 'number', ['number'], [789]]);
+      });
+
+      it('releaseCPointer', function() {
+        var ccall = sandbox.mock(rocky.module)
+          .expects('ccall').twice();
+        sandbox.mock(rocky.module.Runtime)
+          .expects('removeFunction').twice()
+          .onCall(0).returns(456)
+          .onCall(1).returns(123);
+
+        font.read_cb = 123;
+        font.get_size_cb = 456;
+        font.resourceId = 789;
+        font.releaseCPointer(321);
+
+        expect(font).to.not.have.a.property('read_cb');
+        expect(font).to.not.have.a.property('get_size_cb');
+        expect(font).to.not.have.a.property('resourceId');
+
+        // sinon's mocks don't seem to work with multiple expectations on the same mock
+        // hence this check afterwards
+        expect(ccall.getCall(0).args).to.eql(
+          ['fonts_unload_custom_font', 'void', ['number'], [321]]);
+        expect(ccall.getCall(1).args).to.eql(
+          ['emx_resources_remove_custom', 'void', ['number'], [789]]);
+      });
+    });
+
+    describe('fonts_load_custom_font', function() {
```

Comment:
Both, `GBitmap` and `GFont` are treated as `resourceObject` in `resourceObjectSetStatusAndCallEvents()` – it's tested for `GBitmap`. I will extract into an isolated test when I do the 3rd resource-based object (PDC or ImageSequence).